### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by Abenezer-Tilahun**
# Description

- In this project, we implemented preserve data for school library.

# General requirements

- There are no linter errors.
- Followed the GitFlow.
- The documentation is professional.

# Ruby requirements

- Followed best practices for Ruby

# Project requirements

- [x] Check the documentation for Ruby's [JSON library](https://ruby-doc.org/stdlib-3.0.0/libdoc/json/rdoc/JSON.html).
- [x] On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json) 
- [x] On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: main.rb [hunk 1]
```diff
@@ -1,7 +1,9 @@
 require_relative './app'
+require './data'
+require 'json'
 
 puts 'Welcome to your school library!!!'
-
+initialize_files
 def main
   app = App.new
   app.start
```
Comment (uwadonat) [inline main.rb:9]: - [x] I getting the error below while running your project. Kindly fix it.
Screenshot from 2022-05-26 14-44-28

**Review by uwadonat — CHANGES_REQUESTED** [review]
Hi **Team,** 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your commit messages are very descriptive ✔️
- Your code is well organized and clean  ✔️
almost
## Required Changes ♻️
_Check the comments under the review._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: main.rb (hunk 1 above)
Comment (vikipretium) [reply main.rb:9]: its fixed @uwadonat, it was the folder name Data and we used data in the path.

**Comment by omar-labana** [comment]
# **Approved ✔️**

Hi team

Congratulations on completing this project! 👏🏼 🎉

You have correctly implemented all of the requirements📃 correctly and without any errors⚠️ or bugs🪲.

Cheers and good luck in your next milestones! 🍀

Feel free to leave any questions or comments in the PR thread(don't forget to tag me 😄) if something is not 100% clear.

---
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**